src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1879,6 +1885,73 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
             return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
         }
 
+        private BoundExpression BindIndexExpression(PrefixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureIndexOperator, diagnostics);
```

Comment:
>CheckFeatureAvailability [](start = 12, length = 24)

It is not clear why are we making this check in the binder when we can do this in the parser and even in the lexer. It looks like other features that could be detected on the token level are checked in the lexer.

Reply:
I think it was a general direction we decided before to report as much as possible in the binder, so that in some of these cases, syntax trees would have no associated errors, and can be saved/reused in IDE. Do you see otherwise?

---
In reply to: [185303112](https://github.com/dotnet/roslyn/pull/26405#discussion_r185303112) [](ancestors = 185303112)

Reply:
added prototype comments.

---
In reply to: [185303112](https://github.com/dotnet/roslyn/pull/26405#discussion_r185303112) [](ancestors = 185303112)